frontend/src/pages/recommendations/RecommendationPage.tsx

```diff
@@ -93,6 +82,7 @@ function RecommendationsPage() {
     }
     locationSearchRef.current = location.search;
 
+    const searchParams = new URLSearchParams(location.search);
```

Comment:
This line looks like a duplicate of the line 64. Is it required?

Reply:
It's not strictly required. But removing it would trigger a warning about a missing dependency for this effect. In the previous version this was avoided by creating this `searchParams` with `useMemo()`, but that does not seem correct either: `searchParams` is mutable (e.g via `searchParams.set("key", "value")`), so memoizing it based on `location.search` could introduce subtle bugs.

That's why I preferred to recreate it explicitly, as it's quite cheap. I will add a comment though.